docker/Dockerfile.amd

```diff
@@ -2,94 +2,170 @@
+    && make -j $(( $(lscpu | awk '/^Socket\(s\)/{ print $2 }') * $(lscpu | awk '/^Core\(s\) per socket/{ print $4 }') ))  \
+    && make install \
+
+# Install OpenMPI
+RUN cd /tmp \
+    && git clone --recursive https://github.com/open-mpi/ompi.git -b ${OMPI_BRANCH} \
+    && cd ompi \
+    && ./autogen.pl \
+    && mkdir build \
+    && cd build \
+    && ../configure --prefix=$OMPI_HOME --with-ucx=$UCX_HOME \
+        --enable-mca-no-build=btl-uct  \
+        --without-verbs \
+        --with-pmix  \
+        --enable-mpi \
+        --enable-mpi-fortran=yes \
+        --disable-debug \
+    && make -j $(( $(lscpu | awk '/^Socket\(s\)/{ print $2 }') * $(lscpu | awk '/^Core\(s\) per socket/{ print $4 }') )) \
+    && make install
+
+# Cleanup
+RUN rm -rf /tmp/ompi && rm -rf /tmp/ucx
+
+
+# Adding OpenMPI and UCX to Environment
+ENV PATH=$OMPI_HOME/bin:$UCX_HOME/bin:$PATH \
+    LD_LIBRARY_PATH=$OMPI_HOME/lib:$UCX_HOME/lib:$LD_LIBRARY_PATH \
+    LIBRARY_PATH=$OMPI_HOME/lib:$UCX_HOME/lib:$LIBRARY_PATH \
+    C_INCLUDE_PATH=$OMPI_HOME/include:$UCX_HOME/include:$C_INCLUDE_PATH \ 
+    CPLUS_INCLUDE_PATH=$OMPI_HOME/include:$UCX_HOME/include:$CPLUS_INCLUDE_PATH \ 
+    CPATH=$OMPI_HOME/include:$UCX_HOME/include:$CPATH \
+    INCLUDE=$OMPI_HOME/include:$UCX_HOME/include:$INCLUDE \
+    PKG_CONFIG_PATH=$OMPI_HOME/lib/pkgconfig:$UCX_HOME/lib/pkgconfig:$PKG_CONFIG_PATH
+
+# Adding environment variable for Running as ROOT and set MCA pml to UCX
+ENV OMPI_ALLOW_RUN_AS_ROOT=1  \
+    OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1 \
+    OMPI_MCA_pml=ucx
+
+
+##############################################################
+# Additional Devito requirements
+##############################################################
+
+# Additional utils and python3
+RUN apt-get update && \
+    apt-get install -y dh-autoreconf python3-venv python3-dev python3-pip
 
 # Build mpi4py against amdclang
-RUN python3 -m venv /venv && \
+RUN which mpicc && \
+    python3 -m venv /venv && \
     /venv/bin/pip install --no-cache-dir --upgrade pip && \
-    CC=$AOMP/bin/amdclang CXX=$AOMP/bin/amdclang++ /venv/bin/pip install --no-cache-dir mpi4py && \
+    /venv/bin/pip install --no-cache-dir mpi4py && \
     rm -rf ~/.cache/pip
 
+
 ########################################################################
 # AOMP for GPUs (OpenMP offloading)
 ########################################################################
 # This will only trigger if arch is aomp since the final stage depends on it
 FROM sdk-base as aomp
 
+# MPI env
+ENV OMPI_CC="amdclang"
+ENV OMPI_CXX="amdclang++"
+
 # Devito env
 ENV DEVITO_ARCH="aomp"
 ENV DEVITO_PLATFORM="amdgpuX"
 ENV DEVITO_LANGUAGE="openmp"
 
+
 ########################################################################
 # HIPCC for GPUs (HIP)
 ########################################################################
 # This will only trigger if arch is hip since the final stage depends on it
 FROM sdk-base as hip
 
+# MPI env
+ENV OMPI_CC="hipcc"
```

Comment:
Note: in my experiments, I haven't tried OMPI_CC=hipcc. 

And even if I did, nothing would have changed.

Remember that the HipCompiler class sets mpicc=hipcc already , so we can actually drop these lines and, if we want to, leave a comment about the above

Reply:
I would keep it for homogeneity in case for example someone uses a fancy version of customcompiler

Reply:
OK